lib/media.js

```diff
@@ -57,50 +52,106 @@ function findTop(str) {
  * @returns {string}
  */
 function splice(str, index, items) {
-  return str.substr(0, index) + _.flattenDeep(items).join('\n') + str.substr(index);
+  return str.substr(0, index) + items + str.substr(index);
+}
+
+/**
+ * Get the contents of a string that come after specified characters.
+ *
+ * @param  {string} str Any string to split
+ * @param  {string} dir The directory the file is in
+ * @return {string}
+ */
+function getFileName(str, dir) {
+  return str.split(dir)[1];
+}
+
+/**
+ * Retrieve the contents of a file
+ *
+ * @param  {array} fileArray   An array of file names
+ * @param  {string} targetDir  The directory to retrieve files from
+ * @param  {string} filePrefix The string that comes right before the file name
+ * @return {Promise}
+ */
+function getContentsOfFiles(fileArray, targetDir, filePrefix) {
+  var allPromises = [],
+    rootDir = process.cwd();
+
+  fileArray.forEach(file => {
+    allPromises.push(files.readFilePromise(path.join(rootDir, targetDir, getFileName(file, filePrefix))));
+  });
+
+  return Promise.all(allPromises)
+    .catch(err => {
+      throw err;
+    });
+}
+
+/**
+ * Concatenate all the styles together in one `<style>` tag
+ * @param  {array} stylesArr The array of all the css files needed for a page
+ * @return {Promise | boolean}
+ */
+function combineStyles(stylesArr) {
+  if (!stylesArr || !stylesArr.length > 0) {
+    return false;
+  }
+
+  // If there are styles, retrieve contents
+  return getContentsOfFiles(stylesArr, 'public/css', '/css/')
+    .then(data => '<style>' + data.join('') + '</style>');
+}
+
+/**
+ * Concatenate all the scripts together in one `<script>` tag
+ *
+ * @param  {array} scriptsArr The array of all the script files needed for a page
+ * @return {Promise | boolean}
+ */
+function combineScripts(scriptsArr) {
+  if (!scriptsArr || !scriptsArr.length) {
```

Comment:
Regarding the array check above, feel like we need to come together as a group for a consensus for this instant and other things we do. Especially if one of our goals is to have our code be uniform enough to look like it was all written by one person.